### PR TITLE
feat: add brand evo font files

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR seeks to 
- add the new brand evolution fonts and its mapping to design system text variants. 
- add `isBrandEvolution` prop to the `Text` component to allow usage

## **Related issues**

Fixes: https://github.com/metamask/metamask-mobile/issues#workspaces/design-system-61e8a2ae77c3a60012e5003c/issues/zh/757

## **Manual testing steps**

1. Go to storybook
2. Go to text component
3. Turn `isBrandEvolution` knob to true

## **Screenshots/Recordings**

| Before  | After  |
|:---:|:---:|
|before|after|

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/component-library/components/Texts/Text/Text.stories.tsx [hunk 1]
```diff
@@ -25,8 +26,69 @@ const TextMeta = {
...
+    ))}
+  </>
+);
+TextVariantsBrandEvolution.argTypes = {
+  variant: { control: false },
+  children: { control: false },
+  isBrandEvolution: { control: false },
+};
+TextVariantsBrandEvolution.args = {
+  color: SAMPLE_TEXT_PROPS.color,
```
Comment (brianacnguyen) [inline app/component-library/components/Texts/Text/Text.stories.tsx:93]: Adding spread of variants before and after brand evolution

**Comment by brianacnguyen** [comment]
@Reviewers: the font files need to be both in the app/fonts folder AND android folders

File: android/link-assets-manifest.json [hunk 2]
```diff
@@ -1,6 +1,70 @@
...
   "migIndex": 1,
   "data": [
+    {
+      "path": "app/fonts/CentraNo1-Black.otf",
+      "sha1": "4c666a5a1b632ff07f2695511ed55a115e7e3b53"
+    },
+    {
+      "path": "app/fonts/CentraNo1-BlackItalic.otf",
+      "sha1": "3c60dff50870abae443b7ab7e6069f30c3b90f49"
+    },
```
Comment (georgewrmarshall) [inline android/link-assets-manifest.json:11]: I understand that we were provided with a comprehensive set of font files, but I think we should limit them to the weights that roughly map to one another, such as regular, medium, semi-bold, and bold. We don't need black, extra-bold, light, or thin, so I suggest removing these weights. They weren't added in the portfolio or extension, and there doesn't seem to be a need for them in mobile either.
```suggestion
```

File: android/link-assets-manifest.json [hunk 3]
```diff
@@ -1,6 +1,70 @@
...
+      "sha1": "c161b047be337f3deb86c863487118851eeaeb27"
+    },
+    {
+      "path": "app/fonts/CentraNo1-Extrabold.otf",
+      "sha1": "65857b94f0f37fdd2ba97bd772d2bdbc2cd48107"
+    },
+    {
+      "path": "app/fonts/CentraNo1-ExtraboldItalic.otf",
+      "sha1": "1fd37371d99f95b2e18a16b4e4a7e95d91abd367"
+    },
```
Comment (georgewrmarshall) [inline android/link-assets-manifest.json:35]: Suggestion: Remove this and associated font files
```suggestion
```

File: android/link-assets-manifest.json (hunk 2 above)
Comment (georgewrmarshall) [inline android/link-assets-manifest.json:11]: I understand that we were provided with a comprehensive set of font files, but I think we should limit them to the weights that roughly map to one another, such as regular, medium, semi-bold, and bold. We don't need black, extra-bold, light, or thin, so I suggest removing these weights. They weren't added in the portfolio or extension, and there doesn't seem to be a need for them in mobile either.
```suggestion
```

File: android/link-assets-manifest.json [hunk 4]
```diff
@@ -1,6 +1,70 @@
...
+      "sha1": "59828b5ecf716bac5f5db474dc910990c6f33853"
+    },
+    {
+      "path": "app/fonts/CentraNo1-Light.otf",
+      "sha1": "2c66c9a5b49c2efd4391afe9cd885100772e484a"
+    },
+    {
+      "path": "app/fonts/CentraNo1-LightItalic.otf",
+      "sha1": "6e0846636be015355def55af1a3e61694433e52d"
+    },
```
Comment (georgewrmarshall) [inline android/link-assets-manifest.json:51]: Suggestion: Remove this and associated font files
```suggestion
```

File: android/link-assets-manifest.json [hunk 5]
```diff
@@ -1,6 +1,70 @@
...
+      "sha1": "c129a146979b642a446574e109b7bcdd8516d5d7"
+    },
+    {
+      "path": "app/fonts/CentraNo1-Thin.otf",
+      "sha1": "71ce7dbf1965ce97518aef399cae87e6fb3f6f01"
+    },
+    {
+      "path": "app/fonts/CentraNo1-ThinItalic.otf",
+      "sha1": "99ef3ef6944d9a84fb686f286359f4600b5ada8a"
+    },
```
Comment (georgewrmarshall) [inline android/link-assets-manifest.json:67]: Suggestion: Remove this and associated font files
```suggestion
```

File: android/link-assets-manifest.json [hunk 6]
```diff
@@ -41,6 +105,34 @@
       "path": "app/fonts/EuclidCircularB-SemiboldItalic.otf",
       "sha1": "77f2e0212711cc8e043acabdc4bd4a264dfeeeb7"
     },
+    {
+      "path": "app/fonts/MMSans-Black.otf",
+      "sha1": "e006ec7e2ec0eeb8200c49d0b25f4e9546e88bc6"
+    },
```
Comment (georgewrmarshall) [inline android/link-assets-manifest.json:111]: Because we are only supporting regular, medium and bold I think we can also remove this and associated font files
```suggestion
```

File: app/component-library/components-temp/TagColored/__snapshots__/TagColored.test.tsx.snap [hunk 7]
```diff
@@ -20,7 +20,7 @@ exports[`TagColored should render TagColored 1`] = `
     style={
       {
         "color": "#6a737d",
-        "fontFamily": "EuclidCircularB-Regular",
+        "fontFamily": "EuclidCircularB-Bold",
```
Comment (georgewrmarshall) [inline app/component-library/components-temp/TagColored/__snapshots__/TagColored.test.tsx.snap:23]: Why was this snap was updated?

File: app/component-library/components/Texts/Text/Text.types.ts [hunk 8]
```diff
@@ -55,5 +55,9 @@ export interface TextProps extends RNTextProps {
   /**
    * Optional prop to add color to text.
    */
-  color?: TextColor;
+  color?: TextColor | string;
```
Comment (georgewrmarshall) [inline app/component-library/components/Texts/Text/Text.types.ts:58]: nit: this seems separate from adding the font files but I don't think it should effect this PR negatively

File: app/component-library/components/Texts/TextWithPrefixIcon/TextWithPrefixIcon.types.ts [hunk 9]
```diff
@@ -19,5 +19,5 @@ export type TextWithPrefixIconStyleSheetVars = Pick<
   TextWithPrefixIconProps,
   'style'
 > & {
-  color: TextColor;
+  color: TextColor | string;
```
Comment (georgewrmarshall) [inline app/component-library/components/Texts/TextWithPrefixIcon/TextWithPrefixIcon.types.ts:22]: This also seems separate from this PR? Is it carried over from the component updates?

File: app/components/UI/Ramp/Views/SendTransaction/__snapshots__/SendTransaction.test.tsx.snap [hunk 10]
```diff
@@ -2235,7 +2235,7 @@ exports[`SendTransaction View renders correctly for token 1`] = `
                                     style={
                                       {
                                         "color": "#141618",
-                                        "fontFamily": "EuclidCircularB-Bold",
+                                        "fontFamily": "EuclidCircularB-Regular",
                                         "fontSize": 18,
                                         "fontWeight": "normal",
                                         "letterSpacing": 0,
```
Comment (georgewrmarshall) [inline app/components/UI/Ramp/Views/SendTransaction/__snapshots__/SendTransaction.test.tsx.snap:2241]: I don't understand why these snapshots are being updated? We shouldn't be changing any of the current font weights? Is this carried over from another PR?

**Review by georgewrmarshall — DISMISSED** [review]
Looking good! I left a couple of suggestions regarding adding only the necessary fonts that map to our current font weights: regular, medium, and bold. Does that make sense? Also, do we have any insights on how font files affect performance on mobile?

Additionally, can you clarify why some snapshots are being updated in the PR? I can't see the connection.

- Checked `Text` component in Storybook locally ✅

Screenshot 2024-09-04 at 4 42 34 PMScreenshot 2024-09-04 at 4 42 22 PM

File: app/component-library/components-temp/TagColored/__snapshots__/TagColored.test.tsx.snap (hunk 7 above)
Comment (brianacnguyen) [reply app/component-library/components-temp/TagColored/__snapshots__/TagColored.test.tsx.snap:23]: This PR also updated to allow custom fontWeight to be added to the `Text` component, so existing use cases with custom fontWeights are now getting the correct font

File: app/component-library/components/Texts/Text/Text.types.ts (hunk 8 above)
Comment (brianacnguyen) [reply app/component-library/components/Texts/Text/Text.types.ts:58]: its an enhancement

File: app/component-library/components/Texts/TextWithPrefixIcon/TextWithPrefixIcon.types.ts (hunk 9 above)
Comment (brianacnguyen) [reply app/component-library/components/Texts/TextWithPrefixIcon/TextWithPrefixIcon.types.ts:22]: yup

File: app/components/UI/Ramp/Views/SendTransaction/__snapshots__/SendTransaction.test.tsx.snap (hunk 10 above)
Comment (brianacnguyen) [reply app/components/UI/Ramp/Views/SendTransaction/__snapshots__/SendTransaction.test.tsx.snap:2241]: This PR also updated to allow custom fontWeight to be added to the Text component, so existing use cases with custom fontWeights are now getting the correct font

File: ios/MetaMask.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -479,7 +492,6 @@
 				57C103F40F394637B5A886FC /* FontAwesome5_Brands.ttf */,
 				EBC2B6371CD846D28B9FAADF /* FontAwesome5_Regular.ttf */,
 				A98DB430A7DA47EFB97EDF8B /* FontAwesome5_Solid.ttf */,
-				EF1C01B7F08047F9B8ADCFBA /* config.json */,
```
Comment (Cal-L) [inline ios/MetaMask.xcodeproj/project.pbxproj:482]: 2 questions:
- Is this removed unintentionally?
- TBH, I don't even think config.json is needed and/or used, which we can probably remove from the root `fonts` directory

File: app/component-library/components/Texts/Text/Text.utils.ts [hunk 12]
```diff
@@ -32,3 +42,24 @@ export const getFontStyleVariant = (
 
   return `EuclidCircularB-${fontSuffix}${styleSuffix}`;
 };
+
+export const FONTFAMILY_BY_TEXTVARIANT: FontFamilyByTextVariant = {
```
Comment (Cal-L) [inline app/component-library/components/Texts/Text/Text.utils.ts:46]: I've noticed some of the const defined in DS combines some of the words that are camel cased in others. To keep consistency, should we maintain the underscores between FONTFAMILY and TEXTVARIANT?

File: ios/MetaMask/Info.plist [hunk 13]
```diff
@@ -72,7 +72,7 @@
 	<key>NSFaceIDUsageDescription</key>
 	<string>$(PRODUCT_NAME) needs to authenticate</string>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string></string>
```
Comment (Cal-L) [inline ios/MetaMask/Info.plist:75]: Can this piece be removed altogether?

File: ios/MetaMask/Info.plist [hunk 14]
```diff
@@ -148,10 +155,5 @@
 	<string>$(MM_MIXPANEL_TOKEN)</string>
 	<key>fox_code</key>
 	<string>$(MM_FOX_CODE)</string>
-	<key>LSApplicationQueriesSchemes</key>
```
Comment (Cal-L) [inline ios/MetaMask/Info.plist:151]: Why was this removed?

**Review by Cal-L — COMMENTED** [review]
left some comments

File: ios/MetaMask.xcodeproj/project.pbxproj (hunk 11 above)
Comment (brianacnguyen) [reply ios/MetaMask.xcodeproj/project.pbxproj:482]: it was not removed intentionally. i'll put it back to be safe

Comment (brianacnguyen) [reply ios/MetaMask.xcodeproj/project.pbxproj:482]: its not removed, just updated below (line 511)

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! 🔥 Approving on behalf of @MetaMask/design-system-engineers but please only merge once @Cal-L has approved as well

**Comment by Cal-L** [comment]
Let's fix lint issue

**Comment by brianacnguyen** [comment]
> Let's fix lint issue

This lint issue is not related to my PR

**Comment by Cal-L** [comment]
Let's fix the Bitrise issue. Looks like an invalid syntax on iOS side

**Review by Cal-L — APPROVED** [review]
Lgtm
